terminal-emulator/src/main/java/com/termux/terminal/KeyHandler.java

```diff
@@ -154,13 +155,29 @@ public static String getCode(int keyCode, int keyMode, boolean cursorApp, boolea
                 return "\015";
 
             case KEYCODE_DPAD_UP:
-                return (keyMode == 0) ? (cursorApp ? "\033OA" : "\033[A") : transformForModifiers("\033[1", keyMode, 'A');
+                if((keyMode & KEYMOD_FUNCTION) != 0) {
+                    return transformForModifiers("\033[5", keyMode, '~');
+                } else {
+                    return (keyMode == 0) ? (cursorApp ? "\033OA" : "\033[A") : transformForModifiers("\033[1", keyMode, 'A');
+                }
             case KEYCODE_DPAD_DOWN:
-                return (keyMode == 0) ? (cursorApp ? "\033OB" : "\033[B") : transformForModifiers("\033[1", keyMode, 'B');
+                if((keyMode & KEYMOD_FUNCTION) != 0) {
+                    return transformForModifiers("\033[6", keyMode, '~');
+                } else {
+                    return (keyMode == 0) ? (cursorApp ? "\033OB" : "\033[B") : transformForModifiers("\033[1", keyMode, 'B');
+                }
             case KEYCODE_DPAD_RIGHT:
-                return (keyMode == 0) ? (cursorApp ? "\033OC" : "\033[C") : transformForModifiers("\033[1", keyMode, 'C');
+                if((keyMode & KEYMOD_FUNCTION) != 0) {
+                    return transformForModifiers("\033[1", keyMode, 'F');
```

Comment:
I don't know if you need the `1` when Ctrl/Alt/Shift is involved but when neither of them is, it will return `\033[1F`, which is wrong?